docs/workflow/blocks/form.rst

```diff
@@ -157,4 +157,190 @@ In the example below, a mapping has a default value, which is saved using the co
+
+You can see this working `here
+<https://app.kendra.io/bandsintown/connect>`_.
+
+.. code-block:: json
+
+  {
+    "type": "form",
+    "label": "Connect",
+    "jsonSchema": {
+        "type": "object",
+        "properties": {
+            "username": {
+                "type": "string",
+                "title": "Username",
+                "default": ""
+            },
+            "password": {
+                "type": "string",
+                "title": "Password",
+                "default": ""
+            }
+        }
+    },
+    "uiSchema": {
+        "password": {
+            "ui:widget": "password"
+        }
+    }
+  }
+
+
+Nested tasks
+------------
+
+You can insert or "nest" another task within a form through the use of uiSchema.
+First specify your nested task's position in the jsonSchema using the property key of your choice.
+Then you can define the schema's content in the form task with your chosen key, 
+within the enclosing “uiSchema” property.
+
+The below example inserts an array of blocks into the form. Each block displays a simple message.
+
+You can see this example working `here
+<https://app.kendra.io/workflow-builder?data=NobwRALgngDgpmAXGAZgewE4FswBowA2AhgEZwFJgDKAriVgJYR5gBWAzmgHZUDGAFnCxEk4GBjTwMEBnHajIRDAHM4EAPrjJcaVAXR4ldhAwMuyljIgEEyAJpoaAAgDu3AOQQn7OHCcR+BnYnAFonJlcGAgInMic0ADcdUwATFLguWKgnGgY+QWEwAF8S-Fz8oRFEcAglVQ0tKWgFXMQXBhT6yhICNF4Aa3l8Hr7BgGFuFAYLarARgflEUEhYWzAsOXYiVUsmG0oAQScN9i2dotwa1coTs4R8K33kA640AJ1jze2EC6vDZFu3121jWdjUTiIr3eGE+pyBRQAuiULnNegMJlgNlxmMhigigA>`_.
+
+.. code-block:: json
+
+  {
+    "type": "form",
+    "label": "Submit",
+    "jsonSchema": {
+        "properties": {
+            "target_property": {
+                "type": "string",
+                "title": "You won't see this - it will be overridden by uiSchema"
+            }
+        }
+    },
+    "uiSchema": {
+        "target_property": {
+            "ui:widget": "blocks",
+            "blocksConfig": {
+                "blocks": [
+                    {
+                        "type": "message",
+                        "title": "A message"
+                    },
+                    {
+                        "type": "message",
+                        "title": "Another message"
+                    },
+                    {
+                        "type": "message",
+                        "title": "Yet another message"
+                    }
+                ]
+            }
+        }
+    }
+}
+
+Nested flows
+------------
+
+If you want to vary a field according to user input, you can achieve this with a nested flow.
+
+Any flow can be nested in any other flow. The nested flow has access to the main flow's data object, 
+context and state - it can use any data stored here. 
+The nested flow's output is then passed to the main flow's data object, just like the output of a conventional task.
+
+As with the main flow, a nested flow can be edited directly with Kendraio App if opened from the Flow Cloud. 
```

Comment:
This is really great stuff.
Unrelated from this PR but @lukestanley I think we need some text along these lines and style as an introduction in the documentation.